torch/distributed/_composable/fsdp/_fsdp_param.py

```diff
@@ -126,6 +126,7 @@ class FSDPParam:
     _sharded_post_forward_param_data: Optional[torch.Tensor]  # 1D
     _sharded_post_forward_param: Optional[nn.Parameter]  # ND
     _unsharded_param: nn.Parameter  # ND
+    unsharded_accumulated_grad: Optional[torch.Tensor]  # ND
```

Comment:
nit: keep this be a private variable to align with all other fields?

Reply:
Sorry, I think the notion of private and public is non-obvious for the `FSDPParam` class. The way I have is is that if `FSDPParamGroup` should access an attribute on `FSDPParam`, then that attribute should be public. (Note that `FSDPParam` will never be accessed publicly by the user, as there is no way to do that today and should not be a way in the future.)

For this attribute and the current implementation, `FSDPParamGroup` needs to check it if it is `None` or not to know if there is an unsharded accumulated gradient to reduce-scatter or not, so I have it as public.